### ISSUE TITLE
Install Kohana dist not from zip, but from composer

### ISSUE THREAD

**Issue by seyfer** [kohana/kohana#60]
Now when download zip archive - there is all Kohana modules in modules folder and almost empty composer.
Better way is install all Kohana modules and core with composer, and place in zip archive only skeleton with application folder.
All Kohana staff installed in `vendor/kohana/core` and `vendor/kohana/modules`
And modules folder in app is for user app modules only.

I'm already made this structure by myself with 3.2 with composer and package declaration with git source. For example

```
{
    "require": {
        "php": ">=5.4",
        "kohana/system": "3.2",
        "kohana/modules/userguide": "3.2",
        "kohana/modules/auth": "3.2",
        "kohana/modules/database": "3.2",
        "kohana/modules/orm": "3.2",
        "kohana/modules/cache": "3.2",
        "kohana/modules/unittest": "3.2",
        "kohana/modules/minion32": "3.2",
        "kohana/modules/minion-log": "3.2",
        "kohana/modules/pagination": "3.2",
...

},
    "extra": {
        "installer-paths": {
            "vendor/kohana/modules/{$name}/": ["type:kohana-module"]
        }
    },
...
"repositories": [
{
            "type": "package",
            "package": {
                "name": "kohana/system",
                "version": "3.2",
                "source": {
                    "type": "git",
                    "url": "https://github.com/kohana/core.git",
                    "reference": "3.2/master"
                }
            }
        },
        {
            "type": "package",
            "package": {
                "name": "kohana/modules/userguide",
                "version": "3.2",
                "source": {
                    "type": "git",
                    "url": "https://github.com/kohana/userguide.git",
                    "reference": "3.2/master"
                }
            }
        },

...
```

And etc. modules. Than i'm just write pathes to modules and system in `index.php`

```
$modules    = 'vendor/kohana/modules';
$system = 'vendor/kohana/core';

```

So i don't have Kohana code cached in my git project and app modules folder only for my app modules, all another i'm install via composer install.
I think this should be by default and this changes need be approved not only for `3.4`, but for `3.2` and `3.3` versions, which in use a lot.

My suggestion is - install all kohana dependencies from packagist with composer. 

(sorry for my English :) )

**Comment by lenton** [comment]
This has already been done for version 3.4: https://github.com/kohana/kohana/pull/41

> I think this should be by default and this changes need be approved not only for 3.4, but for 3.2 and 3.3 versions, which in use a lot.

Once a Kohana version is released we only make bug fixes, so it can't be implemented in 3.2 or 3.3. You'll have to upgrade if you want it :wink:

**Comment by acoulton** [comment]
@seyfer as @lenton says, all kohana packages can already be installed with composer. We've used that to replace the git submodules in kohana/kohana from 3.4, but 3.3 also supports composer for your own projects.

I wasn't sure if you meant the .zip download should also be largely empty, with just the composer.json - if so then I disagree - the .zip is there for the minority of people who don't have and don't want to use composer at all and just want to download and unpack a .zip with everything in it. I've no idea who they are or why they'd want to do that, but there you go....
